src/main/java/io/pinecone/commons/IndexInterface.java

```diff
@@ -266,6 +266,8 @@ default DescribeIndexStatsRequest validateDescribeIndexStatsRequest(Struct filte
         return describeIndexStatsRequest.build();
     }
 
+    T upsert(List<VectorWithUnsignedIndices> vectorList, String namespace);
```

Comment:
The upsert method was not added to the interface even though it was added to the children classes

Reply:
Good catch, thanks.